### PR TITLE
get pid api

### PR DESCRIPTION
**Pull request by PradeepKiruvale**
## Proposed changes

Implement `get_pid` api and re-use it.

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Improvement (general improvements like code refactoring that doesn't explicitly fix a bug or add any new functionality)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Paste Link to the issue
<br/>

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [x] I ran `cargo fmt` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [x] I used `cargo clippy` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: crates/common/tedge_utils/src/pid.rs [hunk 1]
```diff
@@ -0,0 +1,5 @@
+use std::process;
+
+pub fn get_pid() -> u32 {
+    process::id()
+}
```
Comment (didier-wenzek) [inline crates/common/tedge_utils/src/pid.rs:5]: I really see no value in giving a new name to an std function.

File: crates/common/flockfile/src/unix.rs [hunk 2]
```diff
@@ -86,7 +86,7 @@ impl Flockfile {
                 })?;
 
         // Convert the PID to a string
-        let pid_string = format!("{}", Pid::this());
+        let pid_string = format!("{}", get_pid());
```
Comment (didier-wenzek) [inline crates/common/flockfile/src/unix.rs:89]: Indeed, there is no point to use `nix::unistd::Pid`. As the process id is only used to be printed, `std::process::id()` is all what we need.

```suggestion
        let pid_string = format!("{}", std::process::id());
```

**Review by didier-wenzek — CHANGES_REQUESTED** [review]
I don't understand the motivation for that PR.

Except to remove the use of `nix::unistd::Pid`  in `flockfile`. But `std::process::id()` is all what we need.

File: crates/common/flockfile/src/unix.rs (hunk 2 above)
Comment (PradeepKiruvale) [reply crates/common/flockfile/src/unix.rs]: updated

File: crates/common/flockfile/src/unix.rs [hunk 3]
```diff
@@ -173,6 +172,7 @@ mod tests {
 
     use super::*;
     use assert_matches::*;
+    use nix::unistd::Pid;
```
Comment (didier-wenzek) [inline crates/common/flockfile/src/unix.rs:175]: Why not also use `std::process::id()` in the tests?

**Review by didier-wenzek — APPROVED** [review]
Approved

File: crates/common/flockfile/src/unix.rs (hunk 3 above)
Comment (PradeepKiruvale) [reply crates/common/flockfile/src/unix.rs:175]: Thought of having a different way of getting the `pid` for testing purposes.

**Pull request merged**